rust/azure_iot_operations_services/tests/azure_device_registry_client_tests.rs

```diff
@@ -0,0 +1,970 @@
+                                // We're told to stop waiting for notifications
+                                log::info!("[{log_identifier}] Cleanup signal received, stopping notification receiver");
+                                break;
+                            }
+                        }
+                    }
+
+                    // After loop exits for ANY reason:
+                    if !first_notification_sent {
+                        first_notification_notify.notify_one();
+                    }
+
+                    log::info!(
+                        "[{log_identifier}] Device update notification receiver closed with count: {count}"
+                    );
+                    count
+                }
+            });
+
+            let response = azure_device_registry_client
+                .get_device(DEVICE1.to_string(), ENDPOINT1.to_string(), TIMEOUT)
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Get device to update the status: {response:?}");
+
+            let response_during_obs = azure_device_registry_client
+                .update_device_plus_endpoint_status(
+                    DEVICE1.to_string(),
+                    ENDPOINT1.to_string(),
+                    DeviceStatus {
+                        config: Some(StatusConfig {
+                            version: response.specification.version,
+                            error: Some(ConfigError {
+                                message: Some(format!(
+                                    "Random test error for observation of device update {}",
+                                    Uuid::new_v4()
+                                )),
+                                ..ConfigError::default()
+                            }),
+                            ..StatusConfig::default()
+                        }),
+                        endpoints: HashMap::new(),
+                    },
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!(
+                "[{log_identifier}] Updated device response after observation: {response_during_obs:?}",
+            );
+
+            // Wait for first notification with timeout (e.g., 30 seconds)
+            let notification_timeout = Duration::from_secs(30);
+            let notified_future = first_notification_notify.notified();
+            let _ = tokio::time::timeout(notification_timeout, notified_future).await;
+
+            azure_device_registry_client
+                .unobserve_device_update_notifications(
+                    DEVICE1.to_string(),
+                    ENDPOINT1.to_string(),
+                    TIMEOUT,
+                )
+                .await
+                .unwrap();
+            log::info!("[{log_identifier}] Device update unobservation: {:?}", ());
+
+            // Signal the notification receiver to stop waiting
+            cleanup_notify.notify_one();
```

Comment:
sending this means that even if it would have received the second notification, now it won't. So sending the second notification would no longer test anything